wasatch/FeatureIdentificationDevice.py

```diff
@@ -1038,231 +1239,167 @@ def get_line(self, trigger=True):
+    # Temperature
+    # ##########################################################################
 
-        # iterate over the ODD pixels of the spectrum
-        for i in range(1, len(spectrum), 2):
+    def select_adc(self, n: int) -> SpectrometerResponse:
+        log.debug("select_adc -> %d", n)
+        self.settings.state.selected_adc = n
+        result = self._send_code(0xed, n, label="SELECT_ADC")
+        self._get_code(0xd5, wLength=2, label="GET_ADC (throwaway)") # stabilization read
+        return result
 
-            # back-out the incorrectly applied "even" gain and offset
-            old = float(spectrum[i])
-            raw = (old - self.settings.eeprom.detector_offset) / self.settings.eeprom.detector_gain
+    def get_secondary_adc_calibrated(self, raw: float = None) -> SpectrometerResponse:
+        response = SpectrometerResponse()
+        if not self._has_linearity_coeffs():
+            log.debug("secondary_adc_calibrated: no calibration")
+            return SpectrometerResponse(data=None, error_lvl=ErrorLevel.low, error_msg="secondary_adc_calibrated: no calibration")
 
-            # apply the correct "odd" gain and offset
-            spectrum[i] = (raw * self.settings.eeprom.detector_gain_odd) + self.settings.eeprom.detector_offset_odd
+        if raw is None:
+            raw_res = self.get_secondary_adc_raw()
+        if raw_res.data is None:
+            return raw_res
 
-            if i < 5 or i > len(spectrum) - 5:
-                log.debug("  pixel %4d: old %.2f raw %.2f new %.2f", i, old, raw, spectrum[i])
+        raw = float(raw_res.data)
 
-        log.debug("after: %d, %d, %d, %d, %d", spectrum[0], spectrum[1], spectrum[2], spectrum[3], spectrum[4])
+        # use the first 4 linearity coefficients as a 3rd-order polynomial
+        calibrated = float(self.settings.eeprom.linearity_coeffs[0]) \
+                   + float(self.settings.eeprom.linearity_coeffs[1]) * raw \
+                   + float(self.settings.eeprom.linearity_coeffs[2]) * raw * raw \
+                   + float(self.settings.eeprom.linearity_coeffs[3]) * raw * raw * raw
+        log.debug("secondary_adc_calibrated: %f", calibrated)
+        response.transfer_response(raw_res)
+        response.data = calibrated
+        return response
 
-        return True
+    def get_secondary_adc_raw(self) -> SpectrometerResponse:
+        # flip to secondary ADC if needed
+        if self.settings.state.selected_adc is None or self.settings.state.selected_adc != 1:
+            self.select_adc(1)
+
+        value = self._get_code(0xd5, wLength=2, label="GET_ADC", lsb_len=2) & 0xfff
+        log.debug("secondary_adc_raw: 0x%04x", value)
+        return value
 
     ##
-    # If a spectrometer has bad_pixels configured in the EEPROM, then average
-    # over them in the driver.
+    # Laser temperature conversion doesn't use EEPROM coeffs at all.
+    # Most Wasatch Raman systems use an IPS Wavelength-Stabilized TO-56
```

Comment:
I would change "Most Wasatch Raman systems" to "Wasatch Raman spectrometers with single-mode lasers"